src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -0,0 +1,257 @@
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author J. Henrichs, Bureau of Meteorology
+# -----------------------------------------------------------------------------
+
+'''
+This module contains the GOcean-specific implementation of the ExtractNode.
+It mostly adds code to create a driver program that can read the created
+output files.
+'''
+
+
+from __future__ import absolute_import, print_function
+from psyclone.configuration import Config
+from psyclone.psyir.nodes import ExtractNode
+
+
+class GOceanExtractNode(ExtractNode):
+    '''
+    This is the Gocean-specific implementation of the extract node.
+    It adds a function 'generate_driver' which creates a GOcean-specific
+    stand-alone driver program that can read the created output files.
+
+    :param ast: reference into the fparser2 parse tree corresponding to \
+        this node.
+    :type ast: sub-class of :py:class:`fparser.two.Fortran2003.Base`
+    :param children: the PSyIR nodes that are children of this node.
+    :type children: list of :py:class:`psyclone.psyGen.Node`
+    :param parent: the parent of this node in the PSyIR tree.
+    :type parent: :py:class:`psyclone.psyGen.Node`
+    :param options: a dictionary with options for transformations.
+    :type options: dictionary of string:values or None
+    :param bool options["create_driver"]: If at code creation time a driver \
```

Comment:
Done.

Reply:
Thanks. You're missing a couple of line-continuation slashes in this description.

Reply:
No idea how this happened ... even stranger: when I created the documentation (both html and pdf), it all worked fine (and I made sure to modify the texts so I am certain I was not seeing a cached/outdated copy). I have no idea why, since I know I tested this when I documented the coding style. Anyway, fixed now.